### PR TITLE
Remove OutputFileInits replace EIO Output With GIO

### PR DESCRIPTION
**Pull request by lefticus**
Pull request overview
---------------------

*Squash Merge Highly Recommended*, this comments should suffice.

This PR completes removal of OutputFileInits global variable and move of EIO outputs to use {fmt}

Expected Diffs:

 * Minor spacing differences that actually increase consistency
 * Rounding differences (documented in OutputFiles.unit.cc) that appear to be bugs in the original RoundSigDigits / TrimSigDigits Code.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [X] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [X] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [X] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by lefticus** [comment]
*Squash merge highly recommended*

**Comment by mitchute** [comment]
This all looks good. Diffs are due to minor formatting changes.

@lefticus From the commit log, it looks like ```remove_objexx_gio__4``` merged into ```...gio__6```, and ```6``` merged into ```7```. So, just confirming that ```5``` is dead and can be deleted alongside the others?

**Comment by lefticus** [comment]
@mitchute Correct all of the existing `remove_objexx_gio__*` branches will be dead after this merge. I was trying to figure out the most efficient way to stages this next set of changes. Also, please do a squash merge when it is merged, there are far too many one line "add a missing space" commits that are unnecessary to the history of the project.

**Comment by mitchute** [comment]
Thanks @lefticus. Lots of good stuff here. Merging.

**Comment by Myoldmopar** [comment]
@lefticus one of the commits that was squashed here led to an accidental addition of a newline in the eso and meter output files.  I used Git Bisect to find the bad commit, but since this was all squashed, I can't narrow it down any more than just this PR.  I'll look around in here, but in the meantime do you know off the top of your head exactly what could have caused this?

File: src/EnergyPlus/SimulationManager.cc [hunk 1]
```diff
@@ -1729,11 +1730,7 @@ namespace SimulationManager {
         // na
 
         // SUBROUTINE PARAMETER DEFINITIONS:
-        static ObjexxFCL::gio::Fmt EndOfDataFormat("(\"End of Data\")"); // Signifies the end of the data block in the output file
-        static std::string const ThreadingHeader("! <Program Control Information:Threads/Parallel Sims>, Threading Supported,Maximum Number of "
-                                                 "Threads, Env Set Threads (OMP_NUM_THREADS), EP Env Set Threads (EP_OMP_NUM_THREADS), IDF Set "
-                                                 "Threads, Number of Threads Used (Interior Radiant Exchange), Number Nominal Surfaces, Number "
-                                                 "Parallel Sims");
+        static constexpr auto EndOfDataFormat("End of Data\n"); // Signifies the end of the data block in the output file
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimulationManager.cc:1733]: @lefticus so this `EndOfDataFormat` is used by the eio file, along with the eso and meter (time series) output files (down [here](https://github.com/NREL/EnergyPlus/pull/7746/files#diff-5b4ea1e38e96bd2ee33e0d0cf51d12b0R1882)).  Is there a reason you needed to put a newline here?  It looks like this is the culprit.

**Comment by Myoldmopar** [comment]
It looks like it was specifically this commit where the newline was added: https://github.com/NREL/EnergyPlus/commit/46e06b295dfca24315955ae91a3d3f311e265a1a, though you later made another commit in this branch to fix that line, but still left the newline.  I feel like we may just need to have a separate string for the EIO that includes the newline, and leave one there without a newline for the ESO/MTR files.

**Comment by lefticus** [comment]
I can fix it quickly actually, give me a second.

**Comment by Myoldmopar** [comment]
Thanks!

**Comment by lefticus** [comment]
Committed (to develop) and references you. 

There's a few places where I was inconsistent in handling newlines with the new file IO.

**Pull request merged**